### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by malikhaiderkhan**
In this project we work on:

- to make sure YAGNI, KISS, and DRY rules and Ruby style guide should be followed.

- Follow SOLID principles.

- Pay special attention to the "Single-responsibility Principle".

- Make sure that the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit).

### CODE REVIEW

**Review by Gambit142 — COMMENTED** [review]
Hi @devendra-alt @malikhaiderkhan @Bleu-Yves-Sopoude 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- You have submitted a project, that only @devendra-alt contributed to the code base. The rest of you made changes to the README file which doesn't count as collaboration. Kindly collaborate properly 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: main.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require_relative 'app'
+require_relative 'lib/app'
 
 def main
   app = App.new
```
Comment (thecodechaser) [inline main.rb:35]: - [ ] In the `main.rb` file please also add code that ask the user for the initial choice of options as requested in the project requirements. Currently, that code is present in the `app.rb` file

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @malikhaiderkhan, and Team👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with code refactoring. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by malikhaiderkhan** [comment]
Added options and run methods in main file.

**Review by V-Blaze — APPROVED** [review]
Hi @malikhaiderkhan, @devendra-alt, @Bleu-Yves-Sopoude   

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- Code follows `AGNI, KISS, DRY,` ruby guide💯
-  Readme looks professional👍
-  Nice communication with reviewer🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**